src/model/ItemMovement.js

```diff
@@ -0,0 +1,675 @@
+'use strict';
+
+var utils = require('utils');
+
+module.exports = ItemMovement;
+
+// Defined statuses from the NPC Movement spec
+var MOVE_CB_STATUS = {
+	DIR_CHANGE: 1,
+	NO_PATH_TO_DEST: 2,
+	ARRIVED_NEAR: 3,
+	ARRIVED: 4,
+	STOP_NEW_MOVE: 5,
+	STOP: 6,
+};
+
+
+/**
+ * The generic object to handle movement of an item.
+ *
+ * @param {object} item the item for which movment is being handled
+ */
+function ItemMovement(item) {
+	this.item = item;
+}
+
+
+/**
+ * A helper function to get an offset y value from the item.
+ *
+ * @param {number} y the y value being offset
+ * @param {boolean} positive if the offset is positive or negative
+ * @returns {number} the new offset y value
+ */
+ItemMovement.prototype.offsetY = function offsetY(y, positive) {
+	var yoff = (this.item.y_offset ? this.item.y_offset : 0);
+	if (positive) {
+		return y + yoff;
+	}
+	else {
+		return y - yoff;
+	}
+};
+
+
+/*
+ * A helper function to get the direction an x value is from the item.
+ *
+ * @param {number} targetX the x target value being inspected
+ * @returns {number} 1/-1 depending on the relation between the item.x and
+ *                   targetX
+ */
+ItemMovement.prototype.dirX = function dirX(targetX) {
+	return (this.item.x < targetX ? 1 : -1);
+};
+
+
+/*
+ * A helper function to get the direction an y value is from the item.
+ *
+ * @param {number} targetY the y target value being inspected
+ * @returns {number} 1/-1 depending on the relation between the offset item.y
+ *                   and targetY
+ */
+ItemMovement.prototype.dirY = function dirY(targetY) {
+	return (this.offsetY(this.item.y, false) < targetY ? 1 : -1);
+};
+
+
+/**
+ * Internal Movement handling functions
+ */
+
+/*
+ * Stop existing item movement.
+ *
+ * @param {object} status the status object sent to the movement callback
+ */
+ItemMovement.prototype.stopMove = function stopMove(status) {
+	var fs = false;
+	// First, Halt the timer
+	this.item.cancelGsTimer('movementTimer', true);
+	// clear the path
+	this.path = null;
+	// Notify the callback
+	if (this.callback) {
+		if (this.flags && 'callBackParams' in this.flags) {
+			fs = this.callback.call(this.item, this.flags.callBackParams);
+		}
+		else {
+			fs = this.callback.call(this.item, {status: status});
+		}
+	}
+	return fs;
+};
```

Comment:
It was fullStatus, but it is if for queueChanges we set a full report or not. I can rename it.